Legends-of-Azeroth/Legends-of-Azeroth-Pandaria-5.4.8
# [Core/AreaTriggers] Fix area trigger enter, exit and rest flags

This implements the new `entered` area trigger flag, so expected actions (rested, quest push) are only taken when an area trigger is entered.

The rest system has also been updated to use flags like TCPP/TC, and to simplify calls to `Player::SetRestFlag`.

Steps to verify tavern rest:

1. Create horde character
2. `.debug areatriggers`
3. `.tele tranquillien`
4. Walk into building
5. See area trigger enter message
6. See rested state
7. Leave building
8. See area trigger left (exited) message
9. See rested state has stopped

Steps to verify quest push:

1. Create horde character
2. `.debug areatriggers`
3. `.levelup 79`
4. `.tele org`
5. See area trigger enter message
6. See quest push for `A Personal Summons`
